test/integration/http_integration.cc

```diff
@@ -768,6 +768,84 @@ void HttpIntegrationTest::testGrpcRetry() {
   }
 }
 
+// Verifies that a retry priority can be configured and affect the host selected during retries.
+// The retry priority will always target P1, which would otherwise never be hit due to P0 being
+// healthy.
+void HttpIntegrationTest::testRetryPriority() {
+  const Upstream::PriorityLoad priority_load{0, 100};
+  Upstream::MockRetryPriorityFactory factory(
+      std::make_shared<NiceMock<Upstream::MockRetryPriority>>(priority_load));
+
+  Registry::InjectFactory<Upstream::RetryPriorityFactory> inject_factory(factory);
+
+  envoy::api::v2::route::RouteAction::RetryPolicy retry_policy;
+  retry_policy.mutable_retry_priority()->set_name(factory.name());
+
+  // Add route with custom retry policy
+  config_helper_.addRoute("host", "/test_retry", "cluster_0", false,
+                          envoy::api::v2::route::RouteAction::NOT_FOUND,
+                          envoy::api::v2::route::VirtualHost::NONE, retry_policy);
+
+  // Use load assignments instead of static hosts. Necessary in order to use priorities.
```

Comment:
Do we have a note regarding this somewhere in the doc? I think it will be useful for the operator to be aware of this.

Reply:
It's really just a consequence of the API, so I don't think it's documented anywhere explicitly. Looks like `load_assignment deprecates` the static host list (https://www.envoyproxy.io/docs/envoy/latest/api-v2/api/v2/cds.proto#envoy-api-field-cluster-load-assignment), so I don't think there is much to be gained by documenting the the difference. 

Afaik it's just because priorities were introduced with EDS, and the recently added `load_assignment` brings those features back into the cluster itself.

Reply:
Got it. Thanks for the clarification.